src/distribution/exponential.rs

```diff
@@ -109,6 +109,19 @@ impl ContinuousCDF<f64, f64> for Exp {
             (-self.rate * x).exp()
         }
     }
+
+    /// Calculates the inverse cumulative distribution function.
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// -ln(1 - p) / λ
+    /// ```
+    ///
+    /// where `p` is the probability and `λ` is the rate
+    fn inverse_cdf(&self, p: f64) -> f64 {
+        -(1.0 - p).ln() / self.rate
```

Comment:
To avoid loss of precision when `p` is small, this should use `ln_1p`:
```suggestion
        -(-p).ln_1p() / self.rate
```